File: TestLeetCode/Tree/IsSubTree.cs
public class IsSubtreeSolution {
    public bool IsSubtree(TreeNode root, TreeNode subRoot) {
        if (subRoot == null) return true;
        if (root == null) return false;

        if (IsSameTree(root, subRoot)) {
            return true;
        }
        return  IsSubtree(root.left, subRoot) || IsSubtree(root.right, subRoot);
    }

    bool IsSameTree(TreeNode root, TreeNode subRoot) {
        if (root == null && subRoot == null) return true;

        if (root == null || subRoot == null) return false;
        if (root.val != subRoot.val) return false;

        return IsSameTree(root.left, subRoot.left) && IsSameTree(root.right, subRoot.right);
    }
}

File: python/tree/is_sub_tree.py
from python.common import TreeNode
from python.tree.is_same_tree import SameTree


class Subtree:
    def is_subtree(self, root: TreeNode | None, subroot: TreeNode | None) -> bool:
        if not subroot: return True
        if not root: return False
        return SameTree().is_same_tree(root, subroot) or self.is_subtree(root.left, subroot) or self.is_subtree(root.right, subroot)
